### PR TITLE
Make `FixedDecimal`'s round to increment methods experimental

### PR DESCRIPTION
**Pull request by jedel1043**

### CODE REVIEW

File: utils/fixed_decimal/src/decimal.rs [hunk 1]
```diff
@@ -1082,13 +1091,94 @@ impl FixedDecimal {
     /// dec.trunc(0);
     /// assert_eq!("1", dec.to_string());
     /// ```
+    #[cfg(not(feature = "experimental"))]
```
Comment (Manishearth) [inline utils/fixed_decimal/src/decimal.rs:1094]: nit: let's not do this, write a private `trunc_to_increment_inner` function and then call them from both

Comment (jedel1043) [reply utils/fixed_decimal/src/decimal.rs:1094]: Would doing something like this also work for you?

```rust
pub fn expand(&mut self, position: i16) {
    #[cfg(feature = "experimental")]
    {
        self.expand_to_increment(position, RoundingIncrement::MultiplesOf1)
    }

    #[cfg(not(feature = "experimental"))]
    {
        // Original impl
    }
}
```

Comment (Manishearth) [reply utils/fixed_decimal/src/decimal.rs:1094]: Basically I don't want us to duplicate the impl

Comment (jedel1043) [reply utils/fixed_decimal/src/decimal.rs:1094]: I'll assume that's a yes

Comment (Manishearth) [reply utils/fixed_decimal/src/decimal.rs:1094]: No, that still duplicates the impl, no?

Comment (jedel1043) [reply utils/fixed_decimal/src/decimal.rs:1094]: But the reason why we're making the methods experimental is to benchmark them, so we need to preserve the original implementation somehow, right?

Comment (Manishearth) [reply utils/fixed_decimal/src/decimal.rs:1094]: oh, are these impls distinct?

In that case: we should have internal methods for all of the implementations and use cfgs on thin wrappers

basically we should not do a big cfg/cfg-not pairing for this kind of thing, it's really hard to read. The actual meat of the impl should be in its own method

Comment (jedel1043) [reply utils/fixed_decimal/src/decimal.rs:1094]: That's fair! In that case I'll extract the implementation into an internal method and call that from the public method.

**Review by sffc — APPROVED** [review]
Normally I don't like landing code with `#[cfg(not(...))]` because it is not tested in CI but it's okay in this case since it's between experimental and non-experimental, and we have a path to removing it soon.

**Review by Manishearth — APPROVED** [review]
ideally we'd have RoundingIncrement be internally available (reexported as public in experimental mode) so that we don't need a separate internal function, but this is fine for now

**Pull request merged**